core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -100,15 +123,30 @@ private static Consumer<ErrorResponse> baseNamespaceErrorHandler() {
         case 422:
           throw new RESTException("Unable to process: %s", error.message());
       }
-    };
+
+      super.accept(error);
+    }
   }
 
   /**
    * Request error handler that handles the common cases that are included with all responses, such
    * as 400, 500, etc.
    */
-  public static Consumer<ErrorResponse> defaultErrorHandler() {
-    return error -> {
+  private static class DefaultErrorHandler extends ErrorHandler {
+    private static final ErrorHandler INSTANCE = new DefaultErrorHandler();
+
+    @Override
+    public ErrorResponse parseResponse(int code, String json) {
+      try {
+        return ErrorResponseParser.fromJson(json);
+      } catch (Exception x) {
+        LOG.warn("Unable to parse error response", x);
```

Comment:
This method creates an ErrorResponse even if it can't be parsed. Before an exception would be thrown, and the original error would be lost.